source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,409 @@
+
+   * - ``minSize``
+     - The minimum size of the connection pool, including idle and in-use connections.
+
+   * - ``maxSize``
+     - The maximum size of the connection pool, including idle and in-use connections.
+
+   * - ``size``
+     - The current size of the connection pool, including idle and in-use connections.
+
+   * - ``checkedOutCount``
+     - The current count of connections that are in use.
+
+
+All MXBean instances created by the driver are under the domain
+``"org.mongodb.driver"``.
+
+For more information on the topics discussed in this subsection, see the
+following resources:
+
+- `Oracle Platform MBean Server Reference Documentation <https://docs.oracle.com/en/java/javase/16/management/overview-java-se-monitoring-and-management.html#GUID-F7B9AB8A-F5A8-472A-AEC6-93B5B7FBE7CE>`__
+- `Oracle MXBean Documentation <https://docs.oracle.com/javase/tutorial/jmx/mbeans/mxbeans.html>`__
+- `Oracle MBean Documentation <https://docs.oracle.com/javase/tutorial/jmx/mbeans/standard.html>`__
+
+JMX and JConsole Example
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how you can monitor the driver's connection pools using JMX
+and **JConsole**. JConsole is a JMX compliant GUI monitoring tool that comes with
+the Java Platform.
+
+.. note:: Consult the Official JMX and JConsole Documentation
+
+   The descriptions of JMX and JConsole in this example are illustrative
+   rather than a source of truth. For guaranteed up to date information, consult
+   the following official Oracle resources:
+
+   - `JConsole documentation <https://www.oracle.com/technical-resources/articles/java/jconsole.html>`__. 
+   - `JMX documentation <https://docs.oracle.com/javase/tutorial/jmx/index.html>`__
+
+The following code snippet adds a ``JMXConnectionPoolListener`` to a
+``MongoClient`` instance. The code then starts a JMX connector server exposing
+the platform MBean server on port 9999 for JConsole to connect to.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/JMXMonitoring.java
+   :language: java
+   :dedent:
+   :start-after: start jmx-example
+   :end-before: end jmx-example
+
+The above code snippet should produce output that looks like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Start the connector server...
+   The connector server started.
+   Press <Enter> to stop the server.
+
+Once you have started your server, open JConsole in your terminal using the
+following command:
+
+.. code-block:: shell
+   
+   jconsole
+
+Once JConsole is open, perform the following actions in the GUI:
+
+#. Select the Java process running the above example code.
+#. Press :guilabel:`Insecure Connection` in the warning dialog box.
+#. Click on the :guilabel:`MBeans` tab.
+#. Inspect your connection pool events under the ``"org.mongodb.driver"`` domain.
+
+When you no longer want to inspect your connection pools in JConsole, do the
+following: 
+
+- Exit JConsole by closing the JConsole window
```

Comment:
Nit: Since you number the other sections for steps, you should here too.

Reply:
As these actions are not ordered (they can be performed in any order), I do not believe these should be numbered.